packages/ui-components/src/components/SelectOption/SelectOption.test.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as React from "react"
+import { cleanup, render, screen, waitFor } from "@testing-library/react"
+import userEvent from "@testing-library/user-event"
+import { Select } from "../Select/Select.component"
+import { SelectOption } from "../SelectOption/SelectOption.component"
+
+globalThis.console.warn = vi.fn()
+describe("SelectOption", () => {
+  afterEach(() => {
+    cleanup()
+    vi.clearAllMocks()
+  })
+
+  test("renders a SelectOption", async () => {
+    await waitFor(() =>
+      render(
+        <Select>
+          <SelectOption value="Option 1" />
+        </Select>
+      )
+    )
+    expect(console.warn).toHaveBeenCalledTimes(1)
+    const toggle = screen.getByRole("button")
+    expect(toggle).toBeInTheDocument()
+    await waitFor(() => userEvent.click(toggle))
+    expect(screen.getByRole("option")).toBeInTheDocument()
+    expect(screen.getByRole("option")).toHaveTextContent("Option 1")
+  })
+
+  test("renders a SelectOption with label as passed", async () => {
+    await waitFor(() =>
```

Comment:
again no need to do anything right now but I'd say that is an unnecessary usage of `waitFor` in a lot of places which could easily increase the time it takes to run tests.

Reply:
Same, going to resolve them